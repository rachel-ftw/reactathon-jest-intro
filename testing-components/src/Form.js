import React, { Component } from 'react'
import './styles/Form.css'

export default class Form extends Component {
  state = {
    searchTerm: '',
  }

  handleChange = event => this.setState({ searchTerm: event.target.value })

  render() {
    const { handleSubmit } = this.props
    const { searchTerm } = this.state

    return (
      <form className='form' onSubmit={event => handleSubmit(event, searchTerm)}>
        <input
          type="text"
          value={searchTerm}
          onChange={event => this.handleChange(event)}
          placeholder="Search for a Gif"
        />
        <button type="submit">search</button>
      </form>
    )
  }
}
