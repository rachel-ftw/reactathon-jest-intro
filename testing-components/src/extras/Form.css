.form {
  padding-top: 10vh;
}

.form input {
  padding: 10px;
  border-radius: 2px;
  border: 2px solid MediumAquamarine;
  font-size: 1.5em;
  transition: all .3s ease-in-out;
}

.form input:hover {
  background-color: PaleTurquoise;
}

.form button {
  color: #fff;
  padding: 10px;
  margin-left: 10px;
  border: 2px solid MediumAquamarine;
  border-radius: 2px;
  background-color: MediumAquamarine;
  font-size: 1.5em;
  transition: all .3s ease-in-out;
}

.form button:hover {
  background-color: Turquoise;
}
