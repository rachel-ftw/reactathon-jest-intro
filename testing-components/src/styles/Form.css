.form {
  padding-top: 10vh;
}

.form input {
  padding: 10px;
  border-radius: 2px;
  border: 1px solid MediumAquamarine;
  font-size: 1.5em;
  transition: all .3s ;
}

.form button {
  color: #fff;
  padding: 10px;
  margin-left: 10px;
  border: 0px;
  border-radius: 2px;
  background-color: MediumAquamarine;
  font-size: 1.5em;
  transition: all .3s ease-in-out;
}

.form button:hover {
  background-color: CadetBlue;
}
