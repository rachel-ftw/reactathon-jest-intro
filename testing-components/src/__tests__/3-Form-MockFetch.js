import React from 'react'
import { shallow } from 'enzyme'

import Form from '../Form'

describe('<Form />', () => {
  let wrapper, mockFetch,
    url = 'http://adventures.for.me',
    myMock = jest.fn()
  
  beforeEach(() => {
    mockFetch = new myMock()
    wrapper = shallow( <Form handleSubmit={mockFetch} />)
  });

  it('with mocking', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('with no state', () => {
    expect(wrapper.instance().props.handleSubmit).toExist;
    expect(wrapper.state()).toEqual({ searchTerm: '' });

    wrapper.setState({ searchTerm: 'hello' })
    expect(wrapper.state()).toEqual({ searchTerm: "hello" });
  });
})
