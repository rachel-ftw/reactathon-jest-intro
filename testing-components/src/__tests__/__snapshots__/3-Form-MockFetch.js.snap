// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> with mocking 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    handleSubmit={mockConstructor {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Search for a Gif"
          type="text"
          value=""
/>,
        <button
          type="submit"
>
          search
</button>,
      ],
      "className": "form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search for a Gif",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "search",
          "type": "submit",
        },
        "ref": null,
        "rendered": "search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Search for a Gif"
            type="text"
            value=""
/>,
          <button
            type="submit"
>
            search
</button>,
        ],
        "className": "form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search for a Gif",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "search",
            "type": "submit",
          },
          "ref": null,
          "rendered": "search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
